client/src/pages/Signup.js

```diff
@@ -0,0 +1,168 @@
+import { Button, TextField } from "@material-ui/core";
+import { withStyles } from "@material-ui/core/styles"
+import React from "react";
+
+const warningMsg = {
+    emptyFieldError: "Required",
+    passwordError: "Password must contain at least 6 characters",
+    passwordNotMatch: "Confirme password must be matched",
+    emailFormatError: "Please enter a correct email"
+}
+
+const styles = theme => ({
+    root: {
+        alignItems: "center",
+    },
+    container: {
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "center",
+    },
+    textField: {
+        marginTop: theme.spacing(5),
+        width: 300
+    }, 
+    button: {
+        marginTop: theme.spacing(5)
+    }
+  });
+
+class Signup extends React.Component {
+    state = {
+        username : "",
+        email: "",
+        password : "",
+        confirmPW: "",
+        usernameWarning: "",
+        usernameError: false,
+        emailWarning: "",
+        emailError: false,
+        passwordWarning: "",
+        passwordError: false,
+        confirmPWWarning: "",
+        confirmPWError: ""
+    };
+
+    onUsernameChange = (event) => {
+        const empty = event.target.value.length == 0
+        this.setState({
+            username: event.target.value,
+            usernameError: empty, 
+            usernameWarning: empty ? warningMsg.emptyFieldError : ""
+        })
+    }
+
+    onPasswordChange = (event) => {
+        const lengthCheck = event.target.value.length < 6
+        const empty = event.target.value.length == 0 
+
+        this.setState({
+            password: event.target.value,
+            passwordError: lengthCheck || empty, 
+            passwordWarning: empty ? warningMsg.emptyFieldError : (lengthCheck ? warningMsg.passwordError : "")
+        })
+    }
+
+    onEmailChange = (event) => {
+        const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+        const properEmail = !re.test(event.target.value) // regex matching
+        const empty = event.target.value.length == 0 
+        this.setState({
+            email: event.target.value,
+            emailError: empty || properEmail,
+            emailWarning: empty ? warningMsg.emptyFieldError : (properEmail ? warningMsg.emailFormatError : "")
+        })
+    }
+
+    onConfirmPWChange = (event) => {
+        const lengthCheck = event.target.value.length < 6
```

Comment:
you can save `event.target.value` in a variable so you don't need to keep repeating `event.target.value`